src/rendering/renderers/shared/texture/Texture.ts

```diff
@@ -387,40 +385,19 @@ export class Texture extends EventEmitter<{
 }
 
 Texture.EMPTY = new Texture({
-
+    label: 'EMPTY',
 });
 
-Texture.EMPTY.label = 'EMPTY';
 Texture.EMPTY.destroy = NOOP;
 
-// create a white canvas
-const canvas = DOMAdapter.get().createCanvas();
-
-const size = 1;
-
-canvas.width = size;
-canvas.height = size;
-
-const ctx = canvas.getContext('2d');
-
-ctx.fillStyle = '#ffffff';
-ctx.fillRect(0, 0, size, size);
-
-// draw red triangle
-ctx.beginPath();
-ctx.moveTo(0, 0);
-ctx.lineTo(size, 0);
-ctx.lineTo(size, size);
-ctx.closePath();
-ctx.fillStyle = '#ffffff';
-ctx.fill();
-
 Texture.WHITE = new Texture({
-    source: new ImageSource({
-        resource: canvas,
+    source: new BufferImageSource({
```

Comment:
Will BufferImageSource work on canvas?

Reply:
great question - yes we will should be able make it work with `putImageData`